src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -141,7 +141,7 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
                     // If we don't have content, or we are doing Expect 100 Continue, then we can't rely on
                     // this and must send our headers immediately.
 
-                    await _stream.WriteAsync(_sendBuffer.ActiveMemory, requestCancellationSource.Token).ConfigureAwait(false);
+                    await _stream.WriteAsync(_sendBuffer.ActiveMemory, endStream: _expect100ContinueCompletionSource == null, requestCancellationSource.Token).ConfigureAwait(false);
```

Comment:
We are already check that `_expect100ContinueCompletionSource != null` on line 138. So this would never be true as far as I can tell. (unless I miss something)

Reply:
The check is with "or" there: `if (_expect100ContinueCompletionSource != null || _request.Content == null)`, so if we entered because _request.Content == null, then _expect100ContinueCompletionSource can be null

Reply:
I would suggest reorganizing this code a bit. The `_request.Content == null` case doesn't actually do any of the logic below; it's only relevant if we actually have content.

So we should be able to test for `_request.Content == null` up front and handle that case with about two lines of code, with an else clause that handles the case where we do have content, with or without 100-continue.

But if you want to defer than change, that's fine. I just think the code as structured is rather confusing.

Reply:
Makes sense. But let's defer that change and get this in for now.